cornichon-binary/src/main/scala/com/github/agourlay/cornichon/binary/BinaryDSL.scala

```diff
@@ -0,0 +1,38 @@
+package main.scala.com.github.agourlay.cornichon.binary
+
+import com.github.agourlay.cornichon.core.{ CornichonError, Session }
+import com.github.agourlay.cornichon.dsl.ProvidedInstances
+import com.github.agourlay.cornichon.feature.BaseFeature
+import com.github.agourlay.cornichon.http.HttpService
+import com.github.agourlay.cornichon.steps.regular.assertStep.{ AssertStep, Assertion, GenericEqualityAssertion }
+
+import org.apache.pdfbox.pdmodel.PDDocument
+
+trait BinaryDSL extends ProvidedInstances { this: BaseFeature ⇒
+
+  case object BinaryBodyBuilder {
+
+    def asPDF: PdfStepBuilder.type = PdfStepBuilder
```

Comment:
It does not look very easy to extend if needed.
Maybe you can keep it for discoverability, but use type classes behind the hood?

Reply:
Good point about extensibility 👍 

In that case a `Builder` specific to each content is returned therefore I wanted to go with extension method instead.

The same for the `body` in the `HttpDsl` which currently stands for `body.asJson` which will have to be rework in the same fashion...eventually.